cmd/shellNode.go

```
@@ -0,0 +1,225 @@
+		}
+
+		sess := truss.NewAWSSession(region, awsRole)
+
+		var instanceID string
+
+		if strings.HasPrefix(args[0], "ip-") {
+			// Lookup instance-id from Kubernetes if given a node name
+			kubeNode, err := describeKubernetesNode(args[0])
+			if err != nil {
+				return errors.Wrap(err, "Unable to describe Kubernetes node")
+			}
+
+			providerIDParts := strings.Split(kubeNode.Spec.ProviderID, "/")
+			instanceID = providerIDParts[len(providerIDParts)-1]
+		} else if strings.HasPrefix(args[0], "i-") {
+			instanceID = args[0]
+		}
+
+		instance, err := describeInstance(instanceID, sess)
+		if err != nil {
+			return errors.Wrap(err, "Unable to describe EC2 instance")
+		}
+
+		availabilityZone := *instance.Placement.AvailabilityZone
+		hostname := *instance.PrivateDnsName
+
+		publicKey, err := getSSHPublicKey()
+		if err != nil {
+			return err
+		}
+
+		username := "ec2-user"
+
+		sendPublicKey(availabilityZone, instanceID, publicKey, username, sess)
+
+		jump, err := getJump()
+		if err != nil {
+			return errors.Wrap(err, "Unable to find jumpbox in config file")
+		}
+
+		err = execSSHCommand(hostname, username, jump)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+}
+
+func getAWSRegionFromKubeconfig() (string, error) {
+	kubeconfig, err := getKubeconfig()
+	if err != nil {
+		return "", err
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return "", err
+	}
+
+	host := config.Host
+	hostParts := strings.Split(host, ".")
+	region := hostParts[len(hostParts)-4]
+
+	return region, nil
+}
+
+func execSSHCommand(hostname string, username string, jump string) error {
+	sshBinary, err := exec.LookPath("ssh")
+	if err != nil {
+		return errors.Wrap(err, "Unable to locate ssh binary")
+	}
+
+	target := fmt.Sprintf("%s@%s", username, hostname)
+	proxyJump := fmt.Sprintf("ProxyJump=%s", jump)
+	syscall.Exec(sshBinary, []string{"ssh", "-o", proxyJump, target}, os.Environ())
```

Looks like this may return an error that I think is getting swallowed, here